.github/workflows/publish-rock.yaml

```diff
@@ -0,0 +1,61 @@
+name: Publish ROCK
+
+on:
+  workflow_call:
+    inputs:
+      rock-dir:
+        description: "Path to rock directory, i.e. directory containing rockcraft.yaml"
+        required: true
+        type: string
+      path-to-rock-file:
+        description: "Path to .rock file produced"
+        required: true
+        type: string
+      source-branch:
+        description: Github branch to checkout. If blank, it will use default values.
+        default: '' # For default behaviour, see https://github.com/actions/checkout/tree/v4/#usage
+        required: false
+        type: string
```

Comment:
Hey, considering our live discussion too, I went ahead and followed what you 're suggesting here but I kept the `source-branch` input. I 've much thought into it and we had two options:
1. skip checking the repo out and pass the `tag` that will the `commit` information
2. check out the repo inside `publish` job and append the `commit` information

What I ended up with was number 2. Since we 've standardized that published ROCKs [should always come with commit information](https://github.com/canonical/bundle-kubeflow/issues/747#issuecomment-1841137355) and this is a ROCK specific workflow (and not some general publish image one), then I find it absurd to move that workload from this workflow to the user-workflow. In theory, we won't have to publish a ROCK again without its commit appended.